quantus/metrics/robustness_metrics/relative_stability.py

```diff
@@ -0,0 +1,312 @@
+
+        if self.return_aggregate:
+            result = self.aggregate_func(result)
+
+        self.all_results.append(result)
+        self.last_results = [result]
+
+        return result
+
+    def _get_perturbed_inputs(
+        self, x_batch, y_batch, **kwargs
+    ) -> Tuple[np.ndarray, Dict]:
+        """
+        Returns:
+            xs_batch: np.ndarray, 5D tensor of perturbed inputs x_batchj
+            kwargs: Dict, updated kwargs
+        """
+
+        if "xs_batch" in kwargs:
+            xs_batch = kwargs.pop("xs_batch")
+            if len(xs_batch.shape) <= len(x_batch.shape):
+                raise ValueError("xs_batch must have 1 more batch axis than x_batch")
+
+            return xs_batch, kwargs
+
+        if "perturb_func" not in kwargs:
+            print('No "perturb_func" provided, using random noise as default')
+
+        perturb_function: Callable = kwargs.pop(
+            "perturb_func", perturb_func.random_noise
+        )
+
+        xs_batch = []
+        it = range(self.num_perturbations)
+
+        if self.display_progressbar:
+            it = tqdm(it, desc=f"Collecting perturbation for {self.name}")
+
+        for _ in it:
+            xs = perturb_function(x_batch, **kwargs)
+            logits = self.model.predict(xs)
+            labels = np.argmax(logits, axis=1)
+
+            same_label_indexes = np.argwhere(labels == y_batch)
+            xs = xs[same_label_indexes].reshape(-1, *xs.shape[1:])
+            xs_batch.append(xs)
+
+        # pull all new images into 0 axes
+        xs_batch = np.vstack(xs_batch)
+        # drop images, which cause dims not to be divisible
+        xs_batch = xs_batch[: xs_batch.shape[0] // x_batch.shape[0] * x_batch.shape[0]]
+        # make xs_batch have the same shape as x_batch, with new batching axis at 0
+        xs_batch = xs_batch.reshape(-1, *x_batch.shape)
+        return xs_batch, kwargs
+
+    def _get_explanations(
+        self, x_batch, y_batch, xs_batch, **kwargs
+    ) -> Tuple[np.ndarray, np.ndarray, Dict]:
+        """
+        Returns:
+            a_batch: np.ndarray, a 4D tensor, representing explanations for x_batch
+            as_batch: np.ndarray, a 5D tensor, representing explanations for xs_batch
+            kwargs: Dict, updated kwargs
+        """
+        if (
+            "explain_func" in kwargs and ("a_batch" in kwargs or "as_batch" in kwargs)
+        ) or (
+            "explain_func" not in kwargs
+            and ("a_batch" not in kwargs or "as_batch" not in kwargs)
+        ):
+            raise ValueError(
+                "Must provide either explain_func or (a_batch and as_batch)"
+            )
+
+        if "explain_func" in kwargs:
+            explain_func: Callable = kwargs.pop("explain_func")
+
+            a_batch = explain_func(
+                model=self.model.get_model(), inputs=x_batch, targets=y_batch, **kwargs
+            )
+
+            it = xs_batch
+            if self.display_progressbar:
```

Comment:
Where is this initalised?

Reply:
There's a call to init of superclass (quantus.Metric),

    def __init__(self, *args, **kwargs):
        """
        ...
        """
        super().__init__(*args, **kwargs)
        ...

And quantus.Metric initialises this property
https://github.com/understandable-machine-intelligence-lab/Quantus/blob/main/quantus/metrics/base.py#L43

Reply:
there's also a test case for this https://github.com/aaarrti/Quantus/blob/relative-stability/tests/metrics/test_relative_stability.py#L379

Reply:
FYI we are working on another PR that later will update this logic

but for now we want to follow the same syntax as the other metrics, as it needs to be able to be updated explicitly when a user passes the kwargs

I am aware that it is not optimal, and that your suggestion to not explicitly include it works, but for now, pls follow the syntax